### ISSUE TITLE
Reports of issues with TaterTrader

### ISSUE THREAD

**Issue by DeadAirRT** [Vectorial1024/v1024_civilian_fleets#17]
Received a report that there is a bit of an issue with subordinates using tater trader.

Supposedly, if a subordinate finishes it's trade first, it wont be able to trade again until the leader finishes a trade and undocks.

Any assistance looking into this would be greatly appreciated.

**Comment by Vectorial1024** [comment]
I've heard about this on the Civ-Fleet Nexus too, but I have no idea what might be causing this at the moment. Will try to look into this.

**Comment by Vectorial1024** [comment]
After a quick check, maybe the error is located at my side, where I missed `minBuy` and `minSell` param of TaterTrade when copying ommands.

**Comment by Vectorial1024** [comment]
When I look at the debug log, I got this consistent error message in the log:

```
[General] 65439.08 ======================================
[=ERROR=] 65439.08 Error in AI script tatertrade on entity 0x45e1f: Invalid wares attribute value for trade filter: null
* Expression: $warebasket.{$ware}
* Action: <find_buy_offer>, line 1012
[General] 65439.08 ======================================
```

The relevant line in the TaterTrade script was this:

```
<find_buy_offer tradepartner="this.ship" space="$sellspaces.{$sector}" result="$buyers" multiple="true" wares="$warebasket.{$ware}">
```

I would believe that since the trade offer finding failed, the ships cannot find any trades and so nothing could be done.

I do not have Split Vendetta at the moment, not sure if that could be related to the error.

**Comment by AndLindemann** [comment]
I‘ll happily gift you the Split DLC if you want to have it. Just let me know how. 
Talented modders shouldn’t be held back by lack of content. ;)

**Comment by Vectorial1024** [comment]
@AndLindemann thanks for the offer! But I don't intend to get one for now.

Still, I personally believe it would be best if modders handle cases where players do not have DLCs, by e.g. filtering out some null values.

**Comment by DeadAirRT** [comment]
Awesome thanks. I will look into those lines. I know i force dependency on the dlc but it really shouldn't need it, i think the worst that would happen is a debug error saying unknown ware

**Comment by DeadAirRT** [comment]
Seems that i fixed in tater ai script. It was an issue with how the traders decided the amount of money they could spend. Have some new parameters for civilian fleets as well.

**Comment by Vectorial1024** [comment]
OK, I should have patched the new/missing params in cdcce870b3b9106c2de71e0cd4822c995c4fb841 too.

There seems to be no further outstanding problems, so closing this for now.
